modules/renter/hostdb/hostdb.go

```diff
@@ -107,7 +107,9 @@ func newHostDB(g modules.Gateway, cs modules.ConsensusSet, persistDir string, de
 	hdb.hostTree = hosttree.New(hdb.calculateHostWeight)
 
 	// Load the prior persistence structures.
+	hdb.mu.Lock()
 	err = hdb.load()
+	hdb.mu.Unlock()
```

Comment:
If you look at the queue scan function, you see it launches a thread that was triggering a race condition